### ISSUE TITLE
⌨ Improper nested headings on applicant page

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#9729]
## ⌨ Accessibility Issue

The application informaiton has some accordions within its section. Those accordions should be `h3` but are actually `h2`.

### WCAG 2.1 Criteria

- https://www.w3.org/WAI/WCAG22/quickref/?versions=2.1#qr-content-structure-separation-programmatic
- https://www.w3.org/WAI/WCAG22/quickref/?versions=2.1#qr-navigation-mechanisms-descriptive
- https://www.w3.org/WAI/WCAG22/quickref/?versions=2.1#qr-navigation-mechanisms-headings

## 📋 Steps to Reproduce

1. Login as admin `admin@test.com`
2. Navigate to a specific application (pool candidate)
3. Scroll down to the "Application information section"
4. Note that the accordion headings are the same level as the section heading (`h2`)

## 📸 Screenshot

![Screenshot 2024-03-13 082625](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/1338e522-47ec-4b0b-89af-0399ce9703ea)

## 🙋‍♀️ Proposed Solution

Just update the accordions to use `h3`.

## ✅ Acceptance Criteria

- [ ] Application information accordion headings are rendered as `h3`
